spec/abilities/event_ability_spec.rb

```diff
@@ -822,4 +822,24 @@
     end
   end
 
+  context 'as person without roles' do
+    let(:person_without_roles) do
+      person = Fabricate(Person.name.downcase.to_sym)
+      person.update!(primary_group: groups(:top_layer))
+      person
```

Comment:
you might just reduce this to a one liner:

`Fabricate(:person, primary_group: groups(:top_layer)`